src/mainboard/amd/romecrb/src/ioapic.rs

```diff
@@ -0,0 +1,61 @@
+use core::ptr;
+
+fn ioapic_read(ioapic_base: u32, register: u32) -> u32 {
+    unsafe {
+        ptr::write_volatile(ioapic_base as *mut u32, register);
+        ptr::read_volatile((ioapic_base + 10) as *mut u32)
+    }
+}
+
+fn ioapic_write(ioapic_base: u32, register: u32, value: u32) {
+    unsafe {
+        ptr::write_volatile(ioapic_base as *mut u32, register);
+        ptr::write_volatile((ioapic_base + 10) as *mut u32, value);
+    }
+}
+
+fn ioapic_interrupt_count(ioapic_base: u32) -> u32 {
+    let mut ioapic_interrupts = (ioapic_read(ioapic_base, 1) >> 16) & 0xFF;
+    if ioapic_interrupts == 0xFF {
+        ioapic_interrupts = 23;
+    }
+    ioapic_interrupts += 1;
```

Comment:
is this one of those times you could just say 
ioapic_interrupts + 1
and skip the ; and the return?

Reply:
oh and further (I'm learning)
if io_apic_interrupts == 0xff {
23
} else {
ioapic_interrupts + 1
}
?
or return 23; 

just wondering. Then it need not me mut, not that it matters?